dns: Support DNS modification in kernel mode

cathay4t opened the pull request:
Will read and write directly from/to /etc/resolv.conf when querying and
applying.

Integration test case included.

liangwen12year commented on rust/src/lib/nispor/mod.rs:
```diff
@@ -1,4 +1,7 @@
+// SPDX-License-Identifier: Apache-2.0
+
 mod apply;
+mod dns;
```
`cargo-fmt  --check` failed, please move this line after the `mod bond;`.

cathay4t replied:
Fixed.

liangwen12year commented on rust/src/lib/nispor/apply.rs:
```diff
@@ -44,6 +45,10 @@ pub(crate) fn nispor_apply(
+    if merged_state.routes.is_changed() {
+        net_conf.routes = Some(gen_nispor_route_confs(&merged_state.routes)?);
+    }
+
```
We are using Nispor to apply the static route, not sure if this is deviating from the purpose of querying Linux network state for Nispor ?

liangwen12year commented on tests/integration/dns_test.py:
```diff
@@ -617,3 +620,44 @@ def test_purge_dns_full_config(static_dns_with_options):
+            - rotate
+        """
+    )
+    with disable_service("NetworkManager"):
```
What is the reason that we have to disable NetworkManager before applying the DNS in kernel mode? Will `nmstatectl apply --kernel` not work as expected if not disabling NetworkManager ?

cathay4t replied:
Indicate we are not using NetworkManager to apply the network config.

In theory, with NetworkManager started, this kernel mode should also works well. Even not so, it's a bug of NM, not nmstate, hence no test of such use case in nmstate CI.

liangwen12year commented on tests/integration/route_test.py:
```diff
@@ -2090,3 +2092,75 @@ def test_append_route_rule(route_rule_test_env):
+
+@pytest.fixture
+def cleanup_veth1_kernel_mode():
+    with disable_service("NetworkManager"):
```
What is the reason that we have to disable NetworkManager first ?

liangwen12year commented:
I tested, no problem was found. Please resolve the conflict.

cathay4t committed:
dns: Support DNS modification in kernel mode

Will read and write directly from/to /etc/resolv.conf when querying and
applying.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>
